### PR TITLE
Enable more golangci-linters

### PR DESCRIPTION
**Pull request by xrstf**
**What does this PR do / Why do we need it**:
This PR enables more linters to make our software more betterer.

Each linter is enabled in a dedicated commit, so we can judge its impact.

**Does this PR introduce a user-facing change?**:
```release-note
Metric `kubermatic_api_init_node_deployment_failures` was renamed to `kubermatic_api_failed_init_node_deployment_total`.
Metric `kubermatic_cloud_controller_provider_reconciliations_successful` was renamed to `kubermatic_cloud_controller_provider_successful_reconciliations_total`.
```

### CODE REVIEW

File: pkg/provider/cloud/openstack/helper.go [hunk 1]
```diff
@@ -88,9 +88,9 @@ func getNetworkByName(netClient *gophercloud.ServiceClient, name string, isExter
 	}
 
 	candidates := []*NetworkWithExternalExt{}
-	for _, n := range existingNetworks {
+	for i, n := range existingNetworks {
 		if n.External == isExternal {
-			candidates = append(candidates, &n)
+			candidates = append(candidates, &existingNetworks[i])
```
Comment (kron4eg) [inline pkg/provider/cloud/openstack/helper.go:93]: oh wow, this one was dangerous!

Comment (xrstf) [reply pkg/provider/cloud/openstack/helper.go:93]: Not really. `candidates` is only used if its length is 1, and if its length is 1, the loop had only one iteration, so no pointer overwriting could have occured. That was also the only reason this ever worked :D

Comment (xrstf) [reply pkg/provider/cloud/openstack/helper.go:93]: ... so yeah, in that sense the old code was dangerous. ^^

**Comment by xrstf** [comment]
/override pre-kubermatic-test-integration
/override pre-kubermatic-e2e-vsphere-ubuntu-1.21
/override pre-kubermatic-e2e-vsphere-ubuntu-1.21-customfolder 
/override pre-kubermatic-e2e-vsphere-ubuntu-1.21-datastore-cluster 

vCenter is offline.

**Comment by xrstf** [comment]
/override pre-kubermatic-e2e-nutanix-centos-1.22
/override pre-kubermatic-e2e-nutanix-ubuntu-1.22

Nutanix VPN is currently not available.

**Comment by xrstf** [comment]
/override pre-kubermatic-e2e-nutanix-centos-1.22
/override pre-kubermatic-e2e-nutanix-ubuntu-1.22

Nutanix VPN is currently not available.

**Pull request merged**